src/IECoreImage/ImageReader.cpp

```diff
@@ -520,5 +525,10 @@ CompoundObjectPtr ImageReader::readHeader()
 
 	header->members()["channelNames"] = new StringVectorData( cn );
 
+	if( m_implementation->isDeep() )
```

Comment:
@johnhaddon , I set it to only add the "deep" key when it is deep, and then, it will always set the value to `True`.
If you prefer that the "deep" key is always added, with a value of either `True` or `False`, let me know.

Reply:
I reckon it should always add the "deep" key - `if header["deep"].value` is more readable than `if "deep" in header and header["deep"].value`...